File: tasks/set-providers.js
const {getChainAddresses} = require("../scripts/addresses");

module.exports = async function (taskArguments, hre, runSuper) {
    const {ethers, network} = hre;
    const addresses = getChainAddresses(network.config.chainId)
    const {deployer} = await ethers.getNamedSigners();
    const routerFactory = await ethers.getContractFactory('Router', deployer);
    const contract = await routerFactory.attach(addresses.router);

    // Set anyswap
    for (const bridge of addresses.getAllBridgeImpl) {
        console.log(bridge);
        await contract.updateBridgeProvider(bridge.code, bridge.address);
    }

    for (const swapper of addresses.getAllSwapperImpl) {
        console.log(swapper);
        await contract.updateSwapProvider(swapper.code, swapper.address);
    }

    console.log('Router updated');
}

File: scripts/set-providers.py
from brownie import Router, ZeroEx, accounts, Contract

from scripts.src.deploy import from_deployer
from scripts.src.addresses import save_addresses, load_addresses

def main():
    network = 'hardhat'

    address = load_addresses(network)

    router = Router.at(address['router'])

    router.updateSwapProvider(
        address['swapImpl']['zeroex']['code'],
        address['swapImpl']['zeroex']['address'],
        from_deployer)

    router.updateBridgeProvider(
        address['bridgeImpl']['anyswap']['code'],
        address['bridgeImpl']['anyswap']['address'],
        from_deployer)
